clang/include/clang/Basic/Attr.td

```diff
@@ -3327,6 +3330,15 @@ def DiagnoseIf : InheritableAttr {
   let Documentation = [DiagnoseIfDocs];
 }
 
+def DiagnoseSpecializations : InheritableAttr {
+  let Spellings = [Clang<"diagnose_specializations", /*AllowInC*/0>];
+  let Args = [StringArgument<"Message", 1>];
```

Comment:
Do we want to allow the user to decide between warning vs error like we do for `diagnose_if`?

Reply:
The idea of this attribute is that it says "if you specialize this anything could happen, so don't do it". I don't think it makes a ton of sense to give the choice of making this a warning by default on the attribute level.

Reply:
Okay, SGTM!